main.go

```diff
@@ -154,15 +154,17 @@ func runDaemon() error {
 	}
 
 	// saves daemon pid to pidfile.
-	if err := utils.NewPidfile(cfg.Pidfile); err != nil {
-		logrus.Errorf("failed to create pidfile: %s", err)
-		return err
-	}
-	defer func() {
-		if err := os.Remove(cfg.Pidfile); err != nil {
-			logrus.Errorf("failed to delete pidfile: %s", err)
+	if cfg.Pidfile != "" {
```

Comment:
If Pidfile exists, do we need to truncate it's content ?

Reply:
Look at `utils.NewPidfile`, it won't go ahead when pidfile has value, it a common logic most process deal with.